test/unit/views/organisations/dataset-overview.test.js

```diff
@@ -116,6 +116,47 @@ describe('Dataset Overview Page', () => {
     expect(documentWithGeometries.querySelector('script[src="/public/js/map.bundle.js"]')).not.toBeNull()
   })
 
+  it('Does not render a notice when none is provided', () => {
+    expect(document.querySelector('.govuk-notification-banner')).toBeNull()
+  })
+
+  it('Renders a notice when one is provided', () => {
+    const paramsWithNotice = {
+      ...params,
+      notice: {
+        type: 'due',
+        deadline: 'deadline'
+      }
+    }
+
+    const htmlWithNotice = stripWhitespace(nunjucks.render('organisations/dataset-overview.html', paramsWithNotice))
+    const domWithNotice = new jsdom.JSDOM(htmlWithNotice)
+    const documentWithNotice = domWithNotice.window.document
+
+    expect(documentWithNotice.querySelector('.govuk-notification-banner')).not.toBeNull()
+    const banner = documentWithNotice.querySelector('.govuk-notification-banner')
+    expect(banner.classList.contains('govuk-notification-banner--warning')).toBeFalsy()
+    expect(banner.textContent).toContain(`You must update your ${paramsWithNotice.dataset.dataset} dataset by deadline`)
+  })
```

Comment:
```suggestion
  it('Renders a notice when one is provided', () => {
    const paramsWithNotice = {
      ...params,
      notice: {
        type: 'due',
        deadline: 'deadline'
      }
    }

    const htmlWithNotice = stripWhitespace(nunjucks.render('organisations/dataset-overview.html', paramsWithNotice))
    const domWithNotice = new jsdom.JSDOM(htmlWithNotice)
    const documentWithNotice = domWithNotice.window.document
    const banner = documentWithNotice.querySelector('.govuk-notification-banner')

    expect(documentWithNotice.querySelector('.govuk-notification-banner')).not.toBeNull()
    expect(banner.classList.contains('govuk-notification-banner--warning')).toBeFalsy()
    expect(banner.textContent).toContain(`You must update your ${paramsWithNotice.dataset.dataset} dataset by deadline`)
  })
```

minor grouping change